Consider using `require_serial: true`

Ruff already runs on multiple threads, so there is no performance benefit on having pre-commit spawn multiple instances of Ruff.

On top of that the output becomes less readable:

With `require_serial: false`
```
$ pre-commit run -a ruff
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

Found 4 error(s).
misc/gen_types.py:5:1: I001 Import block is un-sorted or un-formatted
misc/license_headers.py:5:1: I001 Import block is un-sorted or un-formatted
packaging/macOS/freeze_program.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/make_installer.py:3:1: I001 Import block is un-sorted or un-formatted
4 potentially fixable with the --fix option.
Found 4 error(s).
.github/scripts/sink_cache.py:7:1: I001 Import block is un-sorted or un-formatted
misc/generate_pyqt.py:5:1: I001 Import block is un-sorted or un-formatted
packaging/macOS/launch_script.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/launch_script.py:3:1: I001 Import block is un-sorted or un-formatted
4 potentially fixable with the --fix option.
Found 2 error(s).
.github/scripts/check_commit_signature.py:10:1: I001 Import block is un-sorted or un-formatted
misc/format_material.py:5:1: I001 Import block is un-sorted or un-formatted
2 potentially fixable with the --fix option.
Found 6 error(s).
.github/scripts/check_newsfragments.py:7:1: I001 Import block is un-sorted or un-formatted
build.py:3:1: I001 Import block is un-sorted or un-formatted
misc/po_to_csv.py:5:1: I001 Import block is un-sorted or un-formatted
misc/releaser.py:31:1: I001 Import block is un-sorted or un-formatted
packaging/wheel/wheel_it.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/freeze_program.py:3:1: I001 Import block is un-sorted or un-formatted
6 potentially fixable with the --fix option.
```

With `require_serial: true`
```
 pre-commit run -a ruff
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

Found 16 error(s).
.github/scripts/check_commit_signature.py:10:1: I001 Import block is un-sorted or un-formatted
.github/scripts/check_newsfragments.py:7:1: I001 Import block is un-sorted or un-formatted
.github/scripts/sink_cache.py:7:1: I001 Import block is un-sorted or un-formatted
build.py:3:1: I001 Import block is un-sorted or un-formatted
misc/format_material.py:5:1: I001 Import block is un-sorted or un-formatted
misc/gen_types.py:5:1: I001 Import block is un-sorted or un-formatted
misc/generate_pyqt.py:5:1: I001 Import block is un-sorted or un-formatted
misc/license_headers.py:5:1: I001 Import block is un-sorted or un-formatted
misc/po_to_csv.py:5:1: I001 Import block is un-sorted or un-formatted
misc/releaser.py:31:1: I001 Import block is un-sorted or un-formatted
packaging/macOS/freeze_program.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/macOS/launch_script.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/wheel/wheel_it.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/freeze_program.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/launch_script.py:3:1: I001 Import block is un-sorted or un-formatted
packaging/windows/make_installer.py:3:1: I001 Import block is un-sorted or un-formatted
16 potentially fixable with the --fix option.
```

--- a/.pre-commit-hooks.yaml
+++ b/.pre-commit-hooks.yaml
@@ -5,6 +5,6 @@
   language: python
   types_or: [python, pyi]
   args: []
-  require_serial: false
+  require_serial: true
   additional_dependencies: []
   minimum_pre_commit_version: "0"

